shared/src/ContentReactor.Shared/BlobRepository/BlobRepository.cs

```diff
@@ -1,132 +1,139 @@
         {
-            var container = BlobClient.GetContainerReference(containerName);
-            var folder = container.GetDirectoryReference(folderName);
-            var blob = folder.GetBlockBlobReference(blobId);
+            var container = BlobServiceClient.GetBlobContainerClient(containerName);
+            await container.CreateIfNotExistsAsync();
+            var blob = container.GetBlockBlobClient(blobId);
 
             // upload blob
-            blob.Properties.ContentType = contentType;
-            await blob.UploadFromStreamAsync(stream);
+            await blob.UploadAsync(stream);
 
             return blob;
         }
 
-        public async Task<CloudBlockBlob> GetBlobAsync(string containerName, string folderName, string blobId, bool includeAttributes = false)
+        public async Task<BlockBlobClient> GetBlobAsync(string containerName, string blobId)
         {
-            var container = BlobClient.GetContainerReference(containerName);
-            var folder = container.GetDirectoryReference(folderName);
-            var blob = folder.GetBlockBlobReference(blobId);
+            var container = BlobServiceClient.GetBlobContainerClient(containerName);
+            await container.CreateIfNotExistsAsync();
+            var blob = container.GetBlockBlobClient(blobId);
 
-            if (! await blob.ExistsAsync())
+            if (!await blob.ExistsAsync())
             {
                 return null;
             }
-            if (includeAttributes)
-            {
-                await blob.FetchAttributesAsync();
-            }
 
             return blob;
         }
 
-        public async Task<IList<CloudBlockBlob>> ListBlobsInFolderAsync(string containerName, string folderName)
+        public async Task<IList<BlobItem>> ListBlobsAsync(string containerName)
         {
-            var container = BlobClient.GetContainerReference(containerName);
-            var folder = container.GetDirectoryReference(folderName);
+            var container = BlobServiceClient.GetBlobContainerClient(containerName);
+            await container.CreateIfNotExistsAsync();
 
-            // list all blobs in folder
-            var blobsInFolder = new List<CloudBlockBlob>();
-            var continuationToken = new BlobContinuationToken();
-            do
+            var blobs = new List<BlobItem>();
+            await foreach (var blob in container.GetBlobsAsync())
             {
-                var currentPage = await folder.ListBlobsSegmentedAsync(true, BlobListingDetails.Metadata, 100, continuationToken, null, null);
-                blobsInFolder.AddRange(currentPage.Results.OfType<CloudBlockBlob>());
-                continuationToken = currentPage.ContinuationToken;
-            } while (continuationToken != null);
+                blobs.Add(blob);
+            }
 
-            return blobsInFolder;
+            return blobs;
         }
```

Comment:
It might be nicer to make a `ToListAsync` extension method on `IAsyncEnumerable<T>` and then just use `container.GetBlobsAsync().ToListAsync()` inline.

Reply:
Resolved